provider/datagen/src/transform/cldr/time_zones/convert.rs

```diff
@@ -138,6 +139,56 @@ impl From<&CldrTimeZonesData<'_>> for ExemplarCitiesV1<'static> {
     }
 }
 
+impl From<&CldrTimeZonesData<'_>> for MetaZonePeriodV1<'static> {
+    fn from(other: &CldrTimeZonesData) -> Self {
+        let data = &other.meta_zone_periods;
+        Self (
+            data
+                .iter()
+                .filter_map(|(key, zone)| {
+                    let mut key = key.clone();
+                    match zone {
+                        ZonePeriod::Region(periods) => {
+                            periods
+                                .iter()
+                                .map(|period| vec![(key.clone(), period.clone())])
+                        }
+                        ZonePeriod::LocationOrSubRegion(place) => {
+                            place
+                                .iter()
+                                .filter_map(move |(inner_key, location_or_subregion)| {
+                                    let mut key = key.clone();
+                                    key.push('/');
+                                    key.push_str(inner_key);
+                                    match location_or_subregion {
+                                        LocationOrSubRegion::Location(periods) => {
+                                            periods
+                                                .iter()
+                                                .map(|period| (key.clone(), period.clone()))
+                                        }
+                                        LocationOrSubRegion::SubRegion(subregion) => {
```

Comment:
Having trouble to convert this part because it will be vector of vector, which is inconsistent with the above part.